ecommerce/frontend/src/components/checkout/addresslist/AddressCard.js

```diff
@@ -0,0 +1,76 @@
+
+export const AddressCard = ({ address, selected = false, onSelect = () => {}, onAddressInfoChange = () => {} }) => {
+    const [editVisible, setEditVisible] = useState(false)
+    const [isLoading, setIsLoading] = useState(false)
+
+    const onDelete = async () => {
+        try {
+            setIsLoading(true)
+            await sleep(2000)
+            onAddressInfoChange()
+            notification.success({ message: 'You have successfully deleted your address!' })
+        } catch (error) {
+            notification.warning({ message: 'There was an error with your request.' })
+        } finally {
+            setIsLoading(false)
+        }
+    }
+
+    return (
+        <div
+            onClick={() => onSelect(address?.id)}
+            className={cls('address-container', {
+                'address-container__selected': selected,
+            })}>
+            <Spin spinning={isLoading}>
+                <div className="address-header">
+                    <input
+                        type="radio"
+                        name="selectedAddress"
+                        checked={selected}
+                        value={address.id}
+                        onChange={val => onSelect(val.target.value)}
+                    />
+                    &nbsp;{address.title}
+                    <div className="address-header-icons">
+                        <AddressModal
+                            onCancel={() => setEditVisible(false)}
+                            onSuccess={() => {
+                                setEditVisible(false)
+                                onAddressInfoChange()
+                            }}
+                            mode="edit"
+                            visible={editVisible}
+                            address={address}
+                        />
+                        <EditOutlined
+                            onClick={() => {
+                                setEditVisible(true)
+                            }}
+                        />{' '}
+                        &nbsp;
+                        <Popconfirm title="Delete this address?" onConfirm={onDelete} okText="Yes" cancelText="No">
+                            <DeleteOutlined />
+                        </Popconfirm>
+                    </div>
+                </div>
+                <div className="address-card">
+                    <div className="address-card-name">{address.name + ' ' + address.surname}</div>
+                    <div className="address-card-address">{address.address}</div>
+                    <div className="address-card-zipcode">{address.zipCode}</div>
+                    <div className="address-card-region">
+                        {address.province + ' / ' + address.city + ' / ' + address.country}
+                    </div>
+                    <div className="address-card-phone">{address.phone.countryCode + address.phone.number}</div>
+                </div>
+            </Spin>
+        </div>
```

Comment:
Can you also add at the bottom "red star next to input fields means they are 'necessary to fill'" for the sake of clearness?

Reply:
added it on address edit form, did not add it on credit card form due to aesthetic reasons